packages/ts-interface-generator/src/interfaceGenerationHelper.ts

```diff
@@ -518,14 +527,24 @@ function buildAST(
 
   const statements: ts.Statement[] = getImports(requiredImports);
 
+  const requiredGenericTypeImports = generateGenericTypeImports(
+    sourceFile,
+    classDeclaration,
+    statements,
+    requiredImports
+  );
+
+  if (requiredGenericTypeImports.length > 0) {
+    statements.push(...requiredGenericTypeImports);
+  }
+
   const myInterface = factory.createInterfaceDeclaration(
     undefined,
-    [
-      factory.createModifier(ts.SyntaxKind.ExportKeyword),
-      factory.createModifier(ts.SyntaxKind.DefaultKeyword),
-    ],
+    classDeclaration.modifiers?.filter(
```

Comment:
I think this does not cover cases like
```ts
class XYZ { ... }
...
export default XYZ;
```
(but I covered both in https://github.com/SAP/ui5-typescript/pull/476)